stm32-modules/heater-shaker/firmware/host_comms_task/uart_hardware.c

```diff
@@ -0,0 +1,108 @@
+#include "stm32f3xx_hal.h"
+#include "uart_hardware.h"
+
+static UART_HandleTypeDef* uart_handle;
+
+static void Error_Handler(void);
+
+void UART_Init(UART_HandleTypeDef *huart)
+{
+  uart_handle = huart;
+
+  huart->Instance        = USARTx;
+
+  huart->Init.BaudRate   = 115200; //to match default STM32CubeIDE VCP setting
+  huart->Init.WordLength = UART_WORDLENGTH_8B;
+  huart->Init.StopBits   = UART_STOPBITS_1;
+  huart->Init.Parity     = UART_PARITY_NONE;
+  huart->Init.HwFlowCtl  = UART_HWCONTROL_NONE;
+  huart->Init.Mode       = UART_MODE_TX_RX;
+  huart->AdvancedInit.AdvFeatureInit = UART_ADVFEATURE_NO_INIT; 
+  if(HAL_UART_DeInit(huart) != HAL_OK)
+  {
+    Error_Handler();
+  }  
+  if(HAL_UART_Init(huart) != HAL_OK)
+  {
+    Error_Handler();
+  }
+}
+
+void UART_DeInit(UART_HandleTypeDef *huart)
+{
+  huart->Instance        = USARTx;
+
+  huart->Init.BaudRate   = 9600;
+  huart->Init.WordLength = UART_WORDLENGTH_8B;
+  huart->Init.StopBits   = UART_STOPBITS_1;
+  huart->Init.Parity     = UART_PARITY_NONE;
+  huart->Init.HwFlowCtl  = UART_HWCONTROL_NONE;
+  huart->Init.Mode       = UART_MODE_TX_RX;
+  huart->AdvancedInit.AdvFeatureInit = UART_ADVFEATURE_NO_INIT; 
+  if(HAL_UART_DeInit(huart) != HAL_OK)
+  {
+    Error_Handler();
+  }  
+}
+
+static void Error_Handler(void)
```

Comment:
We should have a globally-visible `Error_Handler` somewhere; let's not add another one here